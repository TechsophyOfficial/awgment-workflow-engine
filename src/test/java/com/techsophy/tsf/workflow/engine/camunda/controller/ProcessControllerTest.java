//package com.techsophy.tsf.workflow.engine.camunda.controller;
//
//import au.com.origin.snapshots.junit5.SnapshotExtension;
//import com.fasterxml.jackson.core.JsonProcessingException;
//import com.fasterxml.jackson.databind.ObjectMapper;
//import com.fasterxml.jackson.databind.type.TypeFactory;
//import com.github.tomakehurst.wiremock.WireMockServer;
//import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
//import com.techsophy.tsf.workflow.engine.camunda.config.TestAuthorizationExtension;
//import com.techsophy.tsf.workflow.engine.camunda.config.TestKeycloakServerExtension;
//import com.techsophy.tsf.workflow.engine.camunda.config.TestSecurityConfig;
//import com.techsophy.tsf.workflow.engine.camunda.dto.*;
//import com.techsophy.tsf.workflow.engine.camunda.model.ApiResponse;
//import com.techsophy.tsf.workflow.engine.camunda.service.RuntimeProcessService;
//import lombok.Cleanup;
//import lombok.RequiredArgsConstructor;
//import org.camunda.bpm.engine.test.Deployment;
//import org.junit.jupiter.api.*;
//import org.junit.jupiter.api.extension.ExtendWith;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
//import org.springframework.boot.test.context.SpringBootTest;
//import org.springframework.core.io.ClassPathResource;
//import org.springframework.http.MediaType;
//import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.JwtRequestPostProcessor;
//import org.springframework.test.context.ActiveProfiles;
//import org.springframework.test.web.servlet.MockMvc;
//import org.springframework.test.web.servlet.MvcResult;
//import org.springframework.test.web.servlet.RequestBuilder;
//import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
//
//import java.io.IOException;
//import java.io.InputStream;
//import java.io.UnsupportedEncodingException;
//import java.util.Map;
//
//import static au.com.origin.snapshots.SnapshotMatcher.expect;
//import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
//import static com.github.tomakehurst.wiremock.client.WireMock.get;
//import static com.techsophy.tsf.workflow.engine.camunda.constants.CamundaRuntimeConstants.BASE_URL;
//import static com.techsophy.tsf.workflow.engine.camunda.constants.FormConstants.BUSINESS_KEY;
//import static org.assertj.core.api.Assertions.assertThat;
//import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.jwt;
//import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
//
//@AutoConfigureMockMvc
//@ActiveProfiles("test")
//@ExtendWith(SnapshotExtension.class)
//@TestInstance(TestInstance.Lifecycle.PER_CLASS)
//@SpringBootTest(classes = TestSecurityConfig.class)
//@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
//@RequiredArgsConstructor(onConstructor_ = {@Autowired})
//@ExtendWith({TestKeycloakServerExtension.class, TestAuthorizationExtension.class})
//class ProcessControllerTest
//{
//    private static final String COMPLETE_TASK_DATA = "testdata/complete-task.json";
//    private static final String PROCESS_REQUEST_DATA = "testdata/process-request.json";
//    private static final String PROCESS_VARIABLES_DATA = "testdata/processvariables.json";
//    private static final String CONTACT_DETAILS_DATA = "testdata/contact-details.json";
//    private static final String CUSTOMER_DETAILS_DATA = "testdata/business-details.json";
//
//    private static final TypeFactory TYPE_FACTORY = TypeFactory.defaultInstance();
//    private static final JwtRequestPostProcessor USER1_JWT = jwt().jwt(jwt -> jwt.claim("preferred_username", "user1@test.com"));
//
//    private final MockMvc mockMvc;
//    private final ObjectMapper objectMapper;
//    private final RuntimeProcessService runtimeProcessService;
//    private static final WireMockServer GATEWAY_URL=new WireMockServer(6666);
//
//    @BeforeAll
//    void init() throws IOException {
//        GATEWAY_URL.start();
//        @Cleanup InputStream contactDetails = new ClassPathResource(CONTACT_DETAILS_DATA).getInputStream();
//        FormSchema form = this.objectMapper.readValue(contactDetails, FormSchema.class);
//        GATEWAY_URL.stubFor(get("/form-runtime/v1/forms/contactDetails")
//                .willReturn(json(objectMapper.writeValueAsString(Map.of(
//                        "data",form,
//                        "success",true,
//                        "message","Form details detched successfully")))));
//        @Cleanup InputStream businessDetails = new ClassPathResource(CUSTOMER_DETAILS_DATA).getInputStream();
//        FormSchema businsessForm = this.objectMapper.readValue(businessDetails, FormSchema.class);
//        GATEWAY_URL.stubFor(get("/form-runtime/v1/forms/businessDetails")
//                .willReturn(json(objectMapper.writeValueAsString(Map.of(
//                        "data",businsessForm,
//                        "success",true,
//                        "message","Form details detched successfully")))));
//    }
//
//    @AfterAll
//    void destroy(){
//        GATEWAY_URL.resetAll();
//        GATEWAY_URL.stop();
//    }
//
//    public static ResponseDefinitionBuilder json(String apiResponse)
//    {
//        return aResponse().withHeader("Content-Type",MediaType.APPLICATION_JSON_VALUE).withBody(apiResponse);
//    }
//
//    @Test
//    @Order(1)
//    @Deployment(resources = {"testdata/api-controller.bpmn"})
//    void testStartProcess() throws Exception
//    {
//        @Cleanup InputStream stream = new ClassPathResource(PROCESS_REQUEST_DATA).getInputStream();
//        ProcessInstanceDTO process = this.objectMapper.readValue(stream, ProcessInstanceDTO.class);
//
//        RequestBuilder requestbuilder = MockMvcRequestBuilders.post(BASE_URL + "/process-instance/start")
//                .with(USER1_JWT).content(this.objectMapper.writeValueAsBytes(process)).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//
//        ProcessInstanceResponseDTO processResponse = this.getResponse(result, ProcessInstanceResponseDTO.class).getData();
//
//        assertThat(processResponse.getBusinessKey()).isEqualTo(process.getBusinessKey());
//        FormSchemaResponse task =runtimeProcessService.getNextTask(process.getBusinessKey());
//        CommentDTO commentDTO=new CommentDTO(task.getTaskId(),processResponse.getProcessInstanceId(),"Hello");
//
//        RequestBuilder requestbuilderComments = MockMvcRequestBuilders.post(BASE_URL + "/comment/create")
//                .with(USER1_JWT).content(this.objectMapper.writeValueAsBytes(commentDTO)).contentType(MediaType.APPLICATION_JSON);
//        mockMvc.perform(requestbuilderComments).andExpect(status().isOk()).andReturn();
//
//        RequestBuilder requestbuilderGetComments = MockMvcRequestBuilders.get(BASE_URL + "/comment").queryParam(
//                "businessKey",process.getBusinessKey())
//                .with(USER1_JWT).content(this.objectMapper.writeValueAsBytes(commentDTO)).contentType(MediaType.APPLICATION_JSON);
//        mockMvc.perform(requestbuilderGetComments).andExpect(status().isOk()).andReturn();
//
//    }
//
//    private <T> ApiResponse<T> getResponse(MvcResult result, Class<T> type) throws UnsupportedEncodingException, JsonProcessingException
//    {
//        return this.objectMapper.readValue(result.getResponse().getContentAsString(), TYPE_FACTORY.constructParametricType(ApiResponse.class, type));
//    }
//
//    @Test
//    @Order(2)
//    void testNextTask() throws Exception
//    {
//
//
//        RequestBuilder requestbuilder =
//                MockMvcRequestBuilders.get(BASE_URL + "/task/next-task")
//                        .queryParam("businessKey", BUSINESS_KEY)
//                        .with(USER1_JWT).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//        ApiResponse<?> response = this.objectMapper.readValue(result.getResponse().getContentAsString(), ApiResponse.class);
//        FormSchemaResponse form = this.objectMapper.readValue(this.objectMapper.writeValueAsBytes(response.getData()), FormSchemaResponse.class);
//
//        expect(form.getFormContent(), form.getVariables(), form.getTaskName(), form.getFormKey()).orderedJson().toMatchSnapshot();
//    }
//
//    @Test
//    @Order(4)
//    void testExistingProcessInstanceTask() throws Exception
//    {
//        @Cleanup InputStream stream = new ClassPathResource(PROCESS_REQUEST_DATA).getInputStream();
//
//        RequestBuilder requestbuilder = MockMvcRequestBuilders.post(BASE_URL + "/process-instance")
//                .with(USER1_JWT).content(new String(stream.readAllBytes())).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//
//        ApiResponse<?> response = this.objectMapper.readValue(result.getResponse().getContentAsString(), ApiResponse.class);
//        ProcessInstanceResponseDTO processResponse = this.objectMapper.readValue(this.objectMapper.writeValueAsBytes(response.getData()), ProcessInstanceResponseDTO.class);
//
//        assertThat(processResponse.getAlreadyExists()).isTrue();
//    }
//
//    @Test
//    @Order(5)
//    void testCompleteTask() throws Exception
//    {
//        @Cleanup InputStream stream = new ClassPathResource(COMPLETE_TASK_DATA).getInputStream();
//
//        RequestBuilder requestbuilder =
//                MockMvcRequestBuilders.post(BASE_URL + "/task/next-task")
//                        .with(USER1_JWT).content(new String(stream.readAllBytes())).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//        ApiResponse<?> response = this.objectMapper.readValue(result.getResponse().getContentAsString(), ApiResponse.class);
//        FormSchemaResponse form = this.objectMapper.readValue(this.objectMapper.writeValueAsBytes(response.getData()), FormSchemaResponse.class);
//
//        expect(form.getFormContent(), form.getVariables(), form.getTaskName(), form.getFormKey()).toMatchSnapshot();
//    }
//
//    @Test
//    @Order(6)
//    void testUpdateProcessVariables() throws Exception
//    {
//        @Cleanup InputStream stream = new ClassPathResource(PROCESS_VARIABLES_DATA).getInputStream();
//
//        RequestBuilder requestbuilder =
//                MockMvcRequestBuilders.post(BASE_URL + "/task/variables")
//                        .with(USER1_JWT).content(new String(stream.readAllBytes())).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//        ApiResponse<?> response = this.objectMapper.readValue(result.getResponse().getContentAsString(), ApiResponse.class);
//
//        assertThat(response.getSuccess()).isTrue();
//    }
//
//    @Test
//    @Order(7)
//    void testCompleteLastTask() throws Exception
//    {
//        @Cleanup InputStream stream = new ClassPathResource(COMPLETE_TASK_DATA).getInputStream();
//
//        RequestBuilder requestbuilder =
//                MockMvcRequestBuilders.post(BASE_URL + "/task/next-task")
//                        .with(USER1_JWT).content(new String(stream.readAllBytes())).contentType(MediaType.APPLICATION_JSON);
//
//        MvcResult result = mockMvc.perform(requestbuilder).andExpect(status().isOk()).andReturn();
//
//        expect(this.objectMapper.readValue(result.getResponse().getContentAsByteArray(), Object.class)).orderedJson().toMatchSnapshot();
//    }
//
////    @AfterAll
////    void cleanup()
////    {
////        assertThat(formRepository.findAll()).hasSize(2);
////
////        formService.deleteForm("contactDetails");
////
////        assertThat(formRepository.findAll()).hasSize(1);
////
////        formService.deleteForm("businessDetails");
////
////        assertThat(formRepository.findAll()).isEmpty();
////    }
//}
