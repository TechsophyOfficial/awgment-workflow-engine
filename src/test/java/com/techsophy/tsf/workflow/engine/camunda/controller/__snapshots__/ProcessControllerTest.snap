com.techsophy.tsf.workflow.engine.camunda.controller.ProcessControllerTest.testCompleteLastTask=[
  {
    "message": "No Form Available",
    "success": true
  }
]


com.techsophy.tsf.workflow.engine.camunda.controller.ProcessControllerTest.testCompleteTask=[
FormioFormSchema(display=form, components=[FormComponent(label=My Business need loan of , placeholder=Loan Requirement, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=true, pattern=, customMessage=, minLength=null, maxLength=null, min=null, max=null), key=myBusinessNeedLoanOf, fieldKey=null, type=textfield, input=true, defaultValue=, inputType=text, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Enter Your Name, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=, customMessage=, minLength=null, maxLength=null, min=null, max=null), key=enterYourName, fieldKey=null, type=textfield, input=true, defaultValue=, inputType=text, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Number, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=null, customMessage=, minLength=null, maxLength=null, min=10, max=10), key=number, fieldKey=null, type=number, input=true, defaultValue=null, inputType=null, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Email, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=, customMessage=, minLength=null, maxLength=null, min=null, max=null), key=email, fieldKey=null, type=email, input=true, defaultValue=null, inputType=email, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Date Of Birth (As per PAN), placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=true, pattern=null, customMessage=, minLength=null, maxLength=null, min=null, max=null), key=dateOfBirthAsPerPan, fieldKey=null, type=datetime, input=true, defaultValue=, inputType=null, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=PAN, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=true, pattern=[A-Z]{5}[0-9]{4}[A-Z]{1}, customMessage=, minLength=10, maxLength=10, min=null, max=null), key=pan, fieldKey=null, type=textfield, input=true, defaultValue=, inputType=text, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Pincode, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=true, pattern=[0-9]{6}|[0-9]{3}\s[0-9]{3}, customMessage=, minLength=6, maxLength=6, min=null, max=null), key=pincode, fieldKey=null, type=textfield, input=true, defaultValue=, inputType=text, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Receive Loan related update on WhatsApp, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=null, customMessage=, minLength=null, maxLength=null, min=null, max=null), key=receiveLoanRelatedUpdateOnWhatsApp, fieldKey=null, type=checkbox, input=true, defaultValue=false, inputType=checkbox, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=HTML, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=null, customMessage=null, minLength=null, maxLength=null, min=null, max=null), key=html, fieldKey=null, type=htmlelement, input=false, defaultValue=null, inputType=null, data=null, values=null, disableOnInvalid=false, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null), FormComponent(label=Submit, placeholder=, description=, disabled=false, validateOn=change, validate=ValidationPropsSchema(required=false, pattern=null, customMessage=null, minLength=null, maxLength=null, min=null, max=null), key=submit, fieldKey=null, type=button, input=true, defaultValue=null, inputType=null, data=null, values=null, disableOnInvalid=true, action=null, event=null, provider=null, providerOptions=null, rows=null, components=null, storage=null, fileTypes=null, multiple=false, url=null, format=null, documentType=null, hideLabel=false, html=null, title=null, columns=null, prefix=null, suffix=null, mask=false, questions=null)])
{firstName=alex, age=25, lastname=carey}
Business Details
businessDetails
]


com.techsophy.tsf.workflow.engine.camunda.controller.ProcessControllerTest.testNextTask=[
  {
    "components": [
      {
        "defaultValue": "",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "text",
        "key": "myBusinessNeedLoanOf",
        "label": "My Business need loan of ",
        "mask": false,
        "multiple": false,
        "placeholder": "Loan Requirement",
        "type": "textfield",
        "validate": {
          "customMessage": "",
          "pattern": "",
          "required": true
        },
        "validateOn": "change"
      },
      {
        "defaultValue": "",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "text",
        "key": "enterYourName",
        "label": "Enter Your Name",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "textfield",
        "validate": {
          "customMessage": "",
          "pattern": "",
          "required": false
        },
        "validateOn": "change"
      },
      {
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "key": "number",
        "label": "Number",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "number",
        "validate": {
          "customMessage": "",
          "max": 10,
          "min": 10,
          "required": false
        },
        "validateOn": "change"
      },
      {
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "email",
        "key": "email",
        "label": "Email",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "email",
        "validate": {
          "customMessage": "",
          "pattern": "",
          "required": false
        },
        "validateOn": "change"
      },
      {
        "defaultValue": "",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "key": "dateOfBirthAsPerPan",
        "label": "Date Of Birth (As per PAN)",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "datetime",
        "validate": {
          "customMessage": "",
          "required": true
        },
        "validateOn": "change"
      },
      {
        "defaultValue": "",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "text",
        "key": "pan",
        "label": "PAN",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "textfield",
        "validate": {
          "customMessage": "",
          "maxLength": 10,
          "minLength": 10,
          "pattern": "[A-Z]{5}[0-9]{4}[A-Z]{1}",
          "required": true
        },
        "validateOn": "change"
      },
      {
        "defaultValue": "",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "text",
        "key": "pincode",
        "label": "Pincode",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "textfield",
        "validate": {
          "customMessage": "",
          "maxLength": 6,
          "minLength": 6,
          "pattern": "[0-9]{6}|[0-9]{3}\\s[0-9]{3}",
          "required": true
        },
        "validateOn": "change"
      },
      {
        "defaultValue": "false",
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "inputType": "checkbox",
        "key": "receiveLoanRelatedUpdateOnWhatsApp",
        "label": "Receive Loan related update on WhatsApp",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "checkbox",
        "validate": {
          "customMessage": "",
          "required": false
        },
        "validateOn": "change"
      },
      {
        "description": "",
        "disableOnInvalid": false,
        "disabled": false,
        "hideLabel": false,
        "input": false,
        "key": "html",
        "label": "HTML",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "htmlelement",
        "validate": {
          "required": false
        },
        "validateOn": "change"
      },
      {
        "description": "",
        "disableOnInvalid": true,
        "disabled": false,
        "hideLabel": false,
        "input": true,
        "key": "submit",
        "label": "Submit",
        "mask": false,
        "multiple": false,
        "placeholder": "",
        "type": "button",
        "validate": {
          "required": false
        },
        "validateOn": "change"
      }
    ],
    "display": "form"
  },
  {
    "age": 25,
    "firstName": "alex",
    "lastname": "carey"
  },
  "Contact Details",
  "contactDetails"
]